features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularStack.cpp

```diff
@@ -0,0 +1,582 @@
+{
+    CellularSocket *socket = (CellularSocket *)handle;
+    if (!socket || !socket->created) {
+        return NSAPI_ERROR_DEVICE_ERROR;
+    }
+
+    nsapi_size_or_error_t ret_val = NSAPI_ERROR_OK;
+
+    /* Check parameters */
+    if (addr.get_ip_version() == NSAPI_UNSPEC) {
+        return NSAPI_ERROR_DEVICE_ERROR;
+    }
+
+    _at.lock();
+
+    ret_val = socket_sendto_impl(socket, addr, data, size);
+
+    _at.unlock();
+
+    return ret_val;
+}
+
+nsapi_error_t UBLOX_AT_CellularStack::socket_close_impl(int sock_id)
+{
+    _at.lock();
+    _at.cmd_start("AT+USOCL=");
+    _at.write_int(sock_id);
+    _at.cmd_stop();
+    _at.resp_start();
+    _at.resp_stop();
+    _at.unlock();
+
+    return _at.get_last_error();
+}
+
+// Find or create a socket from the list.
+UBLOX_AT_CellularStack::CellularSocket * UBLOX_AT_CellularStack::find_socket(int id)
+{
+    CellularSocket *socket = NULL;
+
+    for (unsigned int x = 0; (socket == NULL) && (x < UBLOX_MAX_SOCKET); x++) {
+        if (_socket) {
+            if (_socket[x]->id == id) {
+                socket = (_socket[x]);
+            }
+        }
+    }
+
+    return socket;
+}
+
+
+// Clear out the storage for a socket
+void UBLOX_AT_CellularStack::clear_socket(CellularSocket * socket)
+{
+    if (socket != NULL) {
+        socket->id       = SOCKET_UNUSED;
+        socket->rx_avail = 0;
+        socket->_cb      = NULL;
+        socket->_data    = NULL;
+        socket->created  = false;
+    }
+}
+
+const char * UBLOX_AT_CellularStack::get_ip_address()
+{
+    _at.lock();
+    _at.cmd_start("AT+UPSND=" PROFILE ",0");
+    _at.cmd_stop();
+
+    _at.resp_start("+UPSND:");
+    if (_at.info_resp()) {
+        _at.skip_param();
+        _at.skip_param();
+        int len = _at.read_string(_ip, NSAPI_IPv4_SIZE-1);
+        if (len == -1) {
+            _ip[0] = '\0';
+            _at.unlock();
+            // no IPV4 address, return
+            return NULL;
+        }
+
+        // in case stack type is not IPV4 only, try to look also for IPV6 address
+        if (_stack_type != IPV4_STACK) {
+            len = _at.read_string(_ip, PDP_IPV6_SIZE-1);
+        }
+    }
+    _at.resp_stop();
+    _at.unlock();
+
+    // we have at least IPV4 address
+    convert_ipv6(_ip);
+
+    return _ip;
+}
+
+nsapi_error_t UBLOX_AT_CellularStack::gethostbyname(const char *host, SocketAddress *address, nsapi_version_t version)
+{
+    char ipAddress[NSAPI_IP_SIZE];
+    nsapi_error_t err = NSAPI_ERROR_NO_CONNECTION;
+
+    _at.lock();
+    if (address->set_ip_address(host)) {
+        err = NSAPI_ERROR_OK;
+    } else {
+        // This interrogation can sometimes take longer than the usual 8 seconds
+        _at.cmd_start("AT+UDNSRN=0,");
+        _at.write_string(host, true);
+        _at.cmd_stop();
+
+        _at.set_at_timeout(60000);
+        _at.resp_start("+UDNSRN:");
+        if (_at.info_resp()) {
+            _at.read_string(ipAddress, sizeof(ipAddress));
+            _at.resp_stop();
```

Comment:
It doesn't change anything here but logically `resp_stop` would be in the same scope with `resp_start`

Reply:
Updated.